docs: document Conversation layer as primary SDK API (#92)
enyst/OpenHands-Tab: docs/agent-sdk-architecture.md

  // Configuration
  setSettings(settings: Partial<OpenHandsSettings>): void;

  // Connection management (remote mode)
  reconnect(): void;
  disconnect(): void;

  // Status
  getStatus(): 'online' | 'offline' | 'connecting';
  getConversationId(): string | undefined;

  // Cleanup
  removeAllListeners(): void;
}
```

### LocalConversation

**File**: `src/conversation/LocalConversation.ts`

**Purpose**: In-memory agent execution within VS Code without an external agent-server.

**Current Status**: ⚠️ **STUB IMPLEMENTATION** - Currently only emits events without actual agent orchestration. Full implementation tracked in issue [to be created].

**Intended Features** (when implemented):
- Runs agent orchestration locally using `AgentOrchestrator`
- EventEmitter-based event dispatching
- Manages tool execution with `LocalWorkspace`
- Terminal events for VS Code integrated terminal
- Full conversation state management
- No external server required (but still VS Code-bound)

**Current Behavior**:
- Accepts user messages and emits MessageEvent
- Does NOT execute LLM calls or tools
- Does NOT run actual agent loop
- Suitable only for testing event flow

> 

Author: ```suggestion
```

context.subscriptions.push({
  dispose: () => {
    conversation.removeAllListeners();
    conversation.disconnect();
  }
});
```

### Pattern 2: Remote Mode with Local Agent-Server

For VS Code usage without an external server, run agent-server on localhost:

```typescript
import { Conversation, isMessageEvent } from '@openhands/agent-sdk-ts';

// Run agent-server locally: uv run agent-server --host 0.0.0.0 --port 3000
const conversation = Conversation({
  serverUrl: 'http://localhost:3000', // connects to local agent-server
  settings: {
    llm: {
      model: 'claude-sonnet-4-20250514',
      usageId: 'local-server',
    },
    secrets: {
      llmApiKey: process.env.ANTHROPIC_API_KEY,
    },
  },
  workspaceRoot: vscode.workspace.workspaceFolders?.[0]?.uri.fsPath || '/workspace',
});

conversation.on('event', (event) => {
  if (isMessageEvent(event) && event.source === 'agent') {
    const text = event.llm_message.content
      .filter(c => c.type === 'text')
      .map(c => c.text)
      .join('');
    console.log('Agent:', text);
  }
});

await conversation.startNewConversation();
await conversation.sendUserMessage('List files in the current directory');
```

> **Note**: LocalConversation (serverUrl: undefined) is currently a stub. For functional local execution, use RemoteConversation with agent-server on localhost.

Author: ```suggestion
**TODO**: LocalConversation (serverUrl: undefined) is currently a stub.
```
